services/provider/server/server.go

```diff
@@ -577,7 +577,292 @@ func (s *OCSProviderServer) RevokeStorageClaim(ctx context.Context, req *pb.Revo
 
 // GetStorageClaim RPC call to get the ceph resources for the StorageClaim.
 func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.StorageClaimConfigRequest) (*pb.StorageClaimConfigResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "not implemented")
+	if req.StorageClaimName == "" || req.StorageConsumerUUID == "" {
+		return nil, fmt.Errorf("invalid request")
+	}
+
+	var extR []*pb.ExternalResource
+
+	storageConsumer, err := s.consumerManager.Get(ctx, req.StorageConsumerUUID)
+	if err != nil {
+		return nil, err
+	}
+
+	switch storageConsumer.Status.State {
+	case ocsv1alpha1.StorageConsumerStateNotEnabled:
+		return nil, status.Errorf(codes.FailedPrecondition, "StorageConsumer is not yet enabled")
+	case ocsv1alpha1.StorageConsumerStateFailed:
+		return nil, status.Errorf(codes.Internal, "StorageConsumer status failed")
+	case ocsv1alpha1.StorageConsumerStateConfiguring:
+		return nil, status.Errorf(codes.Unavailable, "waiting for the resources to be provisioned")
+	case ocsv1alpha1.StorageConsumerStateDeleting:
+		return nil, status.Errorf(codes.NotFound, "StorageConsumer is in deleting phase")
+	case ocsv1alpha1.StorageConsumerStateReady:
+		consumerConfigMap := &v1.ConfigMap{}
+		if storageConsumer.Status.ResourceNameMappingConfigMap.Name == "" {
+			return nil, fmt.Errorf("waiting for ResourceNameMappingConfig to be generated")
+		}
+		consumerConfigMap.Name = storageConsumer.Status.ResourceNameMappingConfigMap.Name
+		consumerConfigMap.Namespace = storageConsumer.Namespace
+		err = s.client.Get(ctx, client.ObjectKeyFromObject(consumerConfigMap), consumerConfigMap)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get %s configMap. %v", consumerConfigMap.Name, err)
+		}
+		kubeResources, err := s.getKubeResources(ctx, storageConsumer)
+		if err != nil {
+			return nil, err
+		}
+
+		consumerConfig := util.WrapStorageConsumerResourceMap(consumerConfigMap.Data)
+
+		storageCluster, err := util.GetStorageClusterInNamespace(ctx, s.client, s.namespace)
+		if err != nil {
+			return nil, err
+		}
+
+		if req.StorageClaimName == fmt.Sprintf("%s-ceph-rbd", storageCluster.Name) {
+			extR, err = s.getStorageClaimConfigBlock(req, kubeResources,
+				storageCluster.Name,
+				consumerConfig.GetCsiRbdProvisionerCephUserName(),
+				consumerConfig.GetCsiRbdNodeCephUserName(),
+				consumerConfig.GetRbdClientProfileName(),
+			)
+			if err != nil {
+				klog.Errorf("failed to get storage claim config. %v", err)
+				return nil, status.Errorf(codes.Internal, "failed to get storage claim config. %v", err)
+			}
+		} else if req.StorageClaimName == fmt.Sprintf("%s-cephfs", storageCluster.Name) {
+			extR, err = s.getStorageClaimConfigCephFs(req, kubeResources,
+				storageCluster.Name,
+				consumerConfig.GetCsiCephFsProvisionerCephUserName(),
+				consumerConfig.GetCsiRbdNodeCephUserName(),
+				consumerConfig.GetCephFsClientProfileName(),
+			)
```

Comment:
```suggestion
			extR, err = s.getStorageClaimConfigCephFs(
			        req, 
			        kubeResources,
				storageCluster.Name,
				consumerConfig.GetCsiCephFsProvisionerCephUserName(),
				consumerConfig.GetCsiRbdNodeCephUserName(),
				consumerConfig.GetCephFsClientProfileName(),
			)
```

Reply:
Make sure to use GetCsiCephFsNodeCephUserName instead of GetCsiRbdNodeCephUserName